src/apps/rubicon/optimism/rubicon.bathToken.token-fetcher.ts

```diff
@@ -0,0 +1,124 @@
+import { Inject } from '@nestjs/common';
+import { BigNumber } from 'ethers/lib/ethers';
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { Register } from '~app-toolkit/decorators';
+import { ContractType } from '~position/contract.interface';
+import { PositionFetcher } from '~position/position-fetcher.interface';
+import { AppTokenPosition } from '~position/position.interface';
+import { Network } from '~types/network.interface';
+
+import { RubiconContractFactory } from '../contracts';
+import { RUBICON_DEFINITION } from '../rubicon.definition';
+
+const appId = RUBICON_DEFINITION.id;
+const groupId = RUBICON_DEFINITION.groups.bathToken.id;
+const network = Network.OPTIMISM_MAINNET;
+
+// Test:
+// http://localhost:5001/apps/rubicon/tokens?groupIds[]=bathToken&network=optimism
+
+@Register.TokenPositionFetcher({ appId, groupId, network })
+export class OptimismRubiconBathTokenTokenFetcher implements PositionFetcher<AppTokenPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(RubiconContractFactory) private readonly rubiconContractFactory: RubiconContractFactory,
+  ) { }
+
+  async getPositions() {
+    // For now, hardcoded in docs
+    const bathTokens = [
+      '0xB0bE5d911E3BD4Ee2A8706cF1fAc8d767A550497',
+      '0x7571CC9895D8E997853B1e0A1521eBd8481aa186',
+      '0xe0e112e8f33d3f437D1F895cbb1A456836125952',
+      '0x60daEC2Fc9d2e0de0577A5C708BcaDBA1458A833',
+      '0xfFBD695bf246c514110f5DAe3Fa88B8c2f42c411',
+      '0xeb5F29AfaaA3f44eca8559c3e8173003060e919f',
+      '0x574a21fE5ea9666DbCA804C9d69d8Caf21d5322b',
```

Comment:
Lowercase the addresses here, and remove any comments you feel are unnecessary below.

Reply:
`    const bathTokens = [
      '0xB0bE5d911E3BD4Ee2A8706cF1fAc8d767A550497'.toLowerCase(),
      '0x7571CC9895D8E997853B1e0A1521eBd8481aa186'.toLowerCase(),
      '0xe0e112e8f33d3f437D1F895cbb1A456836125952'.toLowerCase(),
      '0x60daEC2Fc9d2e0de0577A5C708BcaDBA1458A833'.toLowerCase(),
      '0xfFBD695bf246c514110f5DAe3Fa88B8c2f42c411'.toLowerCase(),
      '0xeb5F29AfaaA3f44eca8559c3e8173003060e919f'.toLowerCase(),
      '0x574a21fE5ea9666DbCA804C9d69d8Caf21d5322b'.toLowerCase(),
    ];
 `
 
 Resolved with this pattern.